src/test/rgw/CMakeLists.txt

```diff
@@ -140,7 +140,8 @@ add_ceph_unittest(unittest_rgw_string)
 # unitttest_rgw_dmclock_queue
 add_executable(unittest_rgw_dmclock_scheduler test_rgw_dmclock_scheduler.cc $<TARGET_OBJECTS:unit-main>)
 add_ceph_unittest(unittest_rgw_dmclock_scheduler)
-target_link_libraries(unittest_rgw_dmclock_scheduler ${rgw_libs} dmclock ${Boost_LIBRARIES})
+target_link_libraries(unittest_rgw_dmclock_scheduler 
+  ${rgw_libs} dmclock ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
```

Comment:
i think we should stop using `Boost_LIBRARIES`wholesale. but instead only cherrypicking the ones we are really using.

@cbodley what do you think?

Reply:
@tchaikov 
I think the most trying is boost_python, since that one requires python libs as well.
So perhaps something like:
 `Boost_LIBRARIES` == everything but boost_python
 `Boost_PYTHON` == `Boost::python ${PYTHON_LIBRARIES}`
Keeps life simple(r), and workable.

Reply:
i'm mainly confused about where this python dependency is coming from; we really shouldn't be pulling that in

Reply:
@cbodley 
Without investigating it too deeply, I would expe3ct the Py deps to come from `Boost_libraries` including also `Boost::python` which in turn needs the general python libs.
But if no `Boost::python` is used here, we could refrain from using it. Like @tchaikov suggests.
It just means running a few builds to see what of Boost is actually needed.

Reply:
I think this part of `src/CMakeLists.txt` is responsable for adding `Boost::python` to `Boost_LIBRARIES`
```
if(WITH_MGR)
  list(APPEND BOOST_COMPONENTS
    python${MGR_PYTHON_VERSION_MAJOR}${MGR_PYTHON_VERSION_MINOR})
endif()
``` 
But why this then only trhows a tantrum here with `unittest_rgw_dmclock_scheduler` is beyond me.

Reply:
To answer myself: `unittest_rgw_dmclock_scheduler` is the only program that is linked using `Boost_LIBRARIES`

Reply:
okay, thanks. does it link if we remove `Boost_LIBRARIES` entirely? ideally these unit tests wouldn't require any additional dependencies, and would everything transitively from rgw_a/radosgw_a

Reply:
It is a bit counter intuitive but even with no `Boost_LIBRARIES` I see libboos*so linked in, but I'm not getting errors.
```
cache /usr/bin/c++   -Wall -Wtype-limits -Wignored-qualifiers -Winit-self -Wpointer-arith -Werror=format-security -fno-strict-aliasing -fsigned-char -Wno-unknown-pragmas -Wno-unused-function -Wno-unused-local-typedef -Wno-varargs -Wno-gnu-designator -Wno-missing-braces -Wno-parentheses -Wno-deprecated-register  -ftemplate-depth-1024 -Wnon-virtual-dtor -Wno-unknown-pragmas -Wno-ignored-qualifiers -Wno-inconsistent-missing-override -Wno-mismatched-tags -Wno-unused-private-field -Wno-address-of-packed-member -DCEPH_DEBUG_MUTEX -fdiagnostics-color=auto -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -O0 -g  -fuse-ld=/usr/local/bin/ld -Wno-unused-command-line-argument  -Wl,--export-dynamic CMakeFiles/unittest_rgw_dmclock_scheduler.dir/test_rgw_dmclock_scheduler.cc.o ../CMakeFiles/unit-main.dir/unit.cc.o  -o ../../../bin/unittest_rgw_dmclock_scheduler  -L/usr/local/lib -Wl,-rpath,/usr/local/lib:/home/jenkins/workspace/ceph-master/build/lib:/usr/local/lib/nss ../../../lib/libgmock_main.a ../../../lib/libgmock.a ../../../lib/libgtest.a -lpthread /usr/local/lib/libldap.so /usr/local/lib/liblber.so ../../../lib/librgw_a.a ../../../lib/libamqp_mock.a ../../../lib/libdmclock.a /usr/local/lib/libboost_coroutine.so /usr/local/lib/libboost_context.so ../../../lib/librados.so.2.0.0 ../../../lib/libcls_otp_client.a ../../../lib/libcls_lock_client.a ../../../lib/libcls_rgw_client.a ../../../lib/libcls_refcount_client.a ../../../lib/libcls_log_client.a ../../../lib/libcls_timeindex_client.a ../../../lib/libcls_version_client.a ../../../lib/libcls_user_client.a ../../../lib/libglobal.a ../../../lib/libceph-common.so.0 ../../../lib/libjson_spirit.a ../../../lib/libcommon_utf8.a ../../../lib/liberasure_code.a ../../../lib/libcrc32.a ../../../lib/libarch.a /usr/local/lib/libboost_thread.so /usr/local/lib/libboost_chrono.so /usr/local/lib/libboost_atomic.so /usr/local/lib/libboost_system.so /usr/local/lib/libboost_random.so /usr/local/lib/libboost_program_options.so /usr/local/lib/libboost_date_time.so /usr/local/lib/libboost_iostreams.so /usr/local/lib/libboost_regex.so /usr/lib/libexecinfo.so -lpthread -lrt /usr/local/lib/libcurl.so /usr/local/lib/libexpat.so /usr/local/lib/libldap.so /usr/local/lib/liblber.so /usr/local/lib/nss/libssl3.so /usr/local/lib/nss/libsmime3.so /usr/local/lib/nss/libnss3.so /usr/local/lib/nss/libnssutil3.so /usr/local/lib/libplds4.so /usr/local/lib/libplc4.so /usr/local/lib/libnspr4.so /usr/local/lib/libcrypto.so /usr/local/lib/liboath.so
```

So I'll update the PR accordingly and see what ceph-jenkins thinks of it.

Reply:
rgw pulls in boost libs from its dependencies (mostly ceph-common)